File: src/main/java/fi/metatavu/metaform/server/persistence/dao/AttachmentDAO.java
package fi.metatavu.metaform.server.persistence.dao;

import java.util.UUID;

import javax.enterprise.context.ApplicationScoped;

import fi.metatavu.metaform.server.persistence.model.Attachment;

/**
 * DAO class for Attachment entity
 * 
 * @author Antti Leppä
 * @author Heikki Kurhinen
 */
@ApplicationScoped
public class AttachmentDAO extends AbstractDAO<Attachment> {
  
  /**
  * Creates new attachment
  *
  * @param id id
  * @param name name
  * @param content content
  * @param contentType contentType
  * @return created attachment
  * @param lastModifier modifier
  */
  public Attachment create(UUID id, String name, byte[] content, String contentType, UUID userId) {
    Attachment attachment = new Attachment();
    attachment.setId(id);
    attachment.setName(name);
    attachment.setContent(content);
    attachment.setContentType(contentType);
    attachment.setUserId(userId);
    return persist(attachment);
  }

  /**
  * Updates name
  *
  * @param name name
  * @return updated attachment
  */
  public Attachment updateName(Attachment attachment, String name) {
    attachment.setName(name);
    return persist(attachment);
  }

  /**
  * Updates content
  *
  * @param content content
  * @return updated attachment
  */
  public Attachment updateContent(Attachment attachment, byte[] content) {
    attachment.setContent(content);
    return persist(attachment);
  }

  /**
  * Updates contentType
  *
  * @param contentType contentType
  * @return updated attachment
  */
  public Attachment updateContentType(Attachment attachment, String contentType) {
    attachment.setContentType(contentType);
    return persist(attachment);
  }

}

File: src/main/kotlin/fi/metatavu/metaform/server/persistence/dao/AttachmentDAO.kt
package fi.metatavu.metaform.server.persistence.dao

import fi.metatavu.metaform.server.persistence.model.Attachment
import java.util.*
import jakarta.enterprise.context.ApplicationScoped

/**
 * DAO class for Attachment entity
 *
 * @author Antti Leppä
 * @author Heikki Kurhinen
 */
@ApplicationScoped
class AttachmentDAO : AbstractDAO<Attachment>() {

  /**
   * Creates new attachment
   *
   * @param id id
   * @param name name
   * @param content content
   * @param contentType contentType
   * @param userId user id
   * @return created attachment
   */
  fun create(
    id: UUID,
    name: String,
    content: ByteArray,
    contentType: String,
    userId: UUID
  ): Attachment {
    val attachment = Attachment()
    attachment.id = id
    attachment.name = name
    attachment.content = content
    attachment.contentType = contentType
    attachment.userId = userId
    return persist(attachment)
  }

  /**
   * Updates name
   *
   * @param attachment attachment
   * @param name name
   * @return updated attachment
   */
  fun updateName(
    attachment: Attachment,
    name: String
  ): Attachment {
    attachment.name = name
    return persist(attachment)
  }

  /**
   * Updates content
   *
   * @param attachment attachment
   * @param content content
   * @return updated attachment
   */
  fun updateContent(
    attachment: Attachment,
    content: ByteArray
  ): Attachment {
    attachment.content = content
    return persist(attachment)
  }

  /**
   * Updates contentType
   *
   * @param attachment attachment
   * @param contentType contentType
   * @return updated attachment
   */
  fun updateContentType(
    attachment: Attachment,
    contentType: String
  ): Attachment {
    attachment.contentType = contentType
    return persist(attachment)
  }
}